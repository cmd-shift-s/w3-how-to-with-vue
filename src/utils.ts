import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
