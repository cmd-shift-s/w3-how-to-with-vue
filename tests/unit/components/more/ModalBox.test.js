import { mount, renderer } from '@/tests/unit/utils'
import ModalBox from '@/components/more/ModalBox.vue'

describe('ModalBox.vue', () => {
  it('render default', () => {
    const wrap = mount(ModalBox)

    expect(wrap.classes()).toContain('modal-box')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#isActive', () => {
    const wrap = mount(ModalBox, {
      propsData: {
        isActive: true
      }
    })

    expect(wrap.classes()).toContain('is-active')

    wrap.vm.close()

    const $emitted = wrap.emitted()
    expect($emitted.close).toBeTruthy()
    expect($emitted['update:isActive']).toBeTruthy()
    expect($emitted['update:isActive'][0]).toEqual([false])

    wrap.setProps({ isActive: false })
    wrap.update()

    expect(wrap.classes()).not.toContain('is-active')
  })

  it('props#isClosable', () => {
    const wrap = mount(ModalBox, {
      propsData: {
        isClosable: false
      }
    })

    wrap.vm.closeDim()
    expect(wrap.emitted().close).toBeFalsy()

    wrap.setProps({ isClosable: true })

    wrap.vm.closeDim()
    expect(wrap.emitted().close).toBeTruthy()
  })
})
