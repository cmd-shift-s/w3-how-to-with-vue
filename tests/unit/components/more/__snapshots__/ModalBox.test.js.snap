// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBox.vue render default 1`] = `
<div class="modal-box">
    <div class="modal-content"></div>
</div>
`;
